src/python/pants/engine/process.py

```diff
@@ -246,6 +336,52 @@ def try_decode(content: bytes) -> str:
         super().__init__("\n".join(err_strings))
 
 
+@rule
+async def run_maybe_coalesced_process_batch(
+    request: MaybeCoalescedProcessBatch,
+    use_coalesced_process_batch: ExperimentalCoalescedProcessBatchingOption,
+) -> FallibleProcessResult:
+    if use_coalesced_process_batch:
+        return await Get(FallibleProcessResult, CoalescedProcessBatch, request)
```

Comment:
From a safety perspective, it might be the case that when batching is disabled, we should instead run file-at-a-time (as that's the easiest way to flush out bugs in having fully declared the dependencies of a particular sub-process). For example: if a file necessary for process `A` is only included by subprocess `B`, then it will be missing from `A`'s cache key, but we will not detect that, because when they do not hit the cache, they will run together here.

Although I suppose that you might still see that bug after a partial-cache hit where you hit the cache for `B` but not `A`, and then fail when you try to run `A` independently...?

Reply:
Actually... hm. Either batched or unbatched could expose bugs, since both the positive and negative case of a file existing might matter.

Reply:
The hope here is the config disabled exhibits the behavior we have today of batching. I think per-file would be worse experience (which is why we removed it in lieu of batching).

That being said I don't have a good way to find bugs other than some clever testing :thinking: